### ISSUE TITLE
Configure components for accessibility from the "outside"

### ISSUE THREAD

**Issue by ajsecord** [material-components/material-components-ios#871]
We have logic in various places that sets text colors so that they are always accessible given a particular background color. Unfortunately, this is nearly impossible to do well from the inside of a component, since it depends on the surrounding UI, especially when background colors are transparent or semi-transparent. It also badly complicates the implementation of the UI component. 

AppBar needs something similar, as does FeatureHighlight—anything that uses text on an arbitrarily-colored background.

We've tried several different experiments, and the theoretical solution we've arrived at is to have a component/class/set of functions that can take a MDCButton (or MDCAppBar, or whatever), inspect it, and set the text color appropriately. If the object itself doesn't have enough information (e.g., the background color is nil and the user's apparent background color comes from some underlying view), then the caller of the code will have to do the right thing and provide that information. We want to avoid "smart" hacks that, for example, inspect the view hierarchy, since they tend to be brittle.

I want to keep this functionality _away_ from concepts of "styling" components or setting "themes" for a particular client app. Those issues are much bigger and the solutions are heavy. We have a fairly focussed need which will be tough enough to get right without increasing scope.

**Comment by romoore** [comment]
I believe this should be handled by color selection within themes.
